cai_causal_graph/identify_utils.py

```diff
@@ -90,46 +92,42 @@ def identify_confounders(graph: CausalGraph, node_1: NodeLike, node_2: NodeLike)
     :return: A list of all confounders between `node_1` and `node_2`.
     """
 
-    def _identify_confounders_no_checks_no_descendant_pruning(clean_graph: CausalGraph, n1: str, n2: str) -> List[str]:
-        """Private function that does not check if DAG or do descendant pruning."""
+    def _identify_confounders_no_checks_no_descendant_pruning_networkx(
+        clean_graph: networkx.DiGraph, n1: str, n2: str
+    ) -> Set[str]:
+        """
+        Private function that does not check if DAG or do descendant pruning.
 
+        This function searches for confounders using the node `n1` as the starting point. Importantly, this
+        does not produce a set of minimal confounders.
+        """
         # create a copy of the provided graph and prune the children of node 1 and node 2
-        final_graph = clean_graph.copy()
-        for child in final_graph.get_children(n1):
-            final_graph.remove_edge(source=n1, destination=child)
-        for child in final_graph.get_children(n2):
-            final_graph.remove_edge(source=n2, destination=child)
+        removed_edges = list()
+        final_graph = clean_graph
+        for child in list(final_graph.successors(n1)):
+            removed_edges.append((n1, child))
```

Comment:
This does not store any edge meta-data, which may be important to the user. Would be useful to either store this metadata, or to store the `Edge` object? The former may be more efficient.

You could just do something like this: `removed_edges_with_meta.append((n1, child, final_graph.get_edge(n1, child).metadata))`. I'm sure there are more efficient ways as well

Reply:
yes, but the meta does not affect the internals of the method. At the start of the user facing method, we do make a copy of the graph, so the original meta is untouched. Hence, I don't need to worry about metadata here :) Does that make sense or did I miss something?

Reply:
Oh right, that makes a lot of sense! For some reason I had thought that we got rid of copying the causal graph entirely within the larger method.

Makes perfect sense, thanks for clarifying.